CodeEditModules/Modules/WorkspaceClient/src/Model/FileItem.swift

```diff
@@ -257,7 +257,13 @@ public extension WorkspaceClient {
             var fileUrl = self.url
             while FileItem.fileManger.fileExists(atPath: fileUrl.path) {
                 let previousName = fileUrl.lastPathComponent
-                fileUrl = fileUrl.deletingLastPathComponent().appendingPathComponent("\(previousName) copy")
+                let previousNameWithoutExtension = (previousName as NSString).deletingPathExtension
+                let filextension = fileUrl.pathExtension
+                let fileName = filextension.isEmpty ?
+                previousName : previousNameWithoutExtension
```

Comment:
somehow this makes no sense in this place previousName and previousNameWithoutExtension should be the same at this point if fileextension is empty

Reply:
Agree. I've made the necessary changes